diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/AnnotationNewLineRuleWarnTest.kt

```diff
@@ -0,0 +1,223 @@
+                    |   
+                    |  @SomeAnnotation @SecondAnnotation 
+                    |  fun someFunc() {
+                    |       val a = 3
+                    |  }
+                    |}
+                """.trimMargin(),
+                LintError(4,3, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test good`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf
+                    |@Inject
+                    |constructor(conf: Int) {
+                    |
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test good 2`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf @Inject constructor(conf: Int) {
+                    |
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test good 3`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf 
+                    |@Inject 
+                    |@SomeAnnotation 
+                    |constructor(conf: Int) {
+                    |
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test bad`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf @Inject @SomeAnnotation constructor(conf: Int) {
+                    |
+                    |}
+                """.trimMargin(),
+                LintError(1,19, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @Inject not on a single line", true),
+                LintError(1,19, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `annotation constructor test bad 2`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf @Inject 
+                    |@SomeAnnotation constructor(conf: Int) {
+                    |
+                    |}
+                """.trimMargin(),
+                LintError(1,19, ruleId, "${Warnings.ANNOTATION_NEW_LINE.warnText()} @SomeAnnotation not on a single line", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.ANNOTATION_NEW_LINE)
+    fun `no warns in func params`() {
+        lintMethod(AnnotationNewLineRule(),
+                """
+                    |public class Conf {
+                    |   fun someFunc(@SomeAnnotation conf: JsonConf, @SecondAnnotation some: Int) {
```

Comment:
need to have a test with two annotations on one function argument

Reply:
added